pkg/instancetype/revision/store.go

```diff
@@ -39,37 +41,97 @@ import (
 	"kubevirt.io/kubevirt/pkg/instancetype/apply"
 	"kubevirt.io/kubevirt/pkg/instancetype/find"
 	preferenceFind "kubevirt.io/kubevirt/pkg/instancetype/preference/find"
+	"kubevirt.io/kubevirt/pkg/pointer"
 	"kubevirt.io/kubevirt/pkg/util"
 )
 
 func (h *revisionHandler) Store(vm *virtv1.VirtualMachine) error {
-	instancetypeRevision, err := h.storeInstancetypeRevision(vm)
+	instancetypeStatusRef, err := h.storeInstancetypeRevision(vm)
 	if err != nil {
 		log.Log.Object(vm).Reason(err).Error("Failed to store ControllerRevision of VirtualMachineInstancetypeSpec for the Virtualmachine.")
 		return err
 	}
 
-	preferenceRevision, err := h.storePreferenceRevision(vm)
+	preferenceStatusRef, err := h.storePreferenceRevision(vm)
 	if err != nil {
 		log.Log.Object(vm).Reason(err).Error("Failed to store ControllerRevision of VirtualMachinePreferenceSpec for the Virtualmachine.")
 		return err
 	}
 
-	return h.patchVM(instancetypeRevision, preferenceRevision, vm)
+	return h.patchVM(instancetypeStatusRef, preferenceStatusRef, vm)
 }
 
-func (h *revisionHandler) storeInstancetypeRevision(vm *virtv1.VirtualMachine) (*appsv1.ControllerRevision, error) {
-	if vm.Spec.Instancetype == nil || vm.Spec.Instancetype.RevisionName != "" {
+func syncStatusWithMatcher(
+	vm *virtv1.VirtualMachine,
+	matcher virtv1.Matcher,
+	statusRef *virtv1.InstancetypeStatusRef,
+	createRevisionFunc func(vm *virtv1.VirtualMachine) (*appsv1.ControllerRevision, error),
+) error {
+	matcherName := matcher.GetName()
+	if matcherName != "" && matcherName != statusRef.Name {
+		statusRef.Name = matcherName
+		statusRef.ControllerRevisionRef = nil
+	}
+
+	matcherKind := matcher.GetKind()
+	if matcherKind != "" && matcherKind != statusRef.Kind {
+		statusRef.Kind = matcherKind
+		statusRef.ControllerRevisionRef = nil
+	}
+
+	matcherInferFromVolume := matcher.GetInferFromVolume()
+	if matcherInferFromVolume != "" && matcherInferFromVolume != statusRef.InferFromVolume {
+		statusRef.InferFromVolume = matcherInferFromVolume
+		statusRef.ControllerRevisionRef = nil
+	}
+
+	matcherInferFromVolumeFailurePolicy := matcher.GetInferFromVolumeFailurePolicy()
+	if matcherInferFromVolumeFailurePolicy != nil && *matcherInferFromVolumeFailurePolicy != *statusRef.InferFromVolumeFailurePolicy {
+		statusRef.InferFromVolumeFailurePolicy = pointer.P(*matcherInferFromVolumeFailurePolicy)
```

Comment:
In this case it is not necesarry to call `statusRef.ControllerRevisionRef = nil`?

Reply:
I don't think so no. The value of `InferFromVolumeFailurePolicy` shouldn't require us to collect a fresh revision.